frontend/src/core/codemirror/language/languages/sql/completion-store.ts

```diff
@@ -33,81 +42,74 @@ class SQLCompletionStore {
+    const builder = new CompletionBuilder();
 
     // When there is only one database, it is the default
-    const defaultDb = connection.databases.find(
-      (db) =>
-        db.name === connection.default_database ||
-        connection.databases.length === 1,
+    const defaultDb = databases.find(
+      (db) => db.name === default_database || databases.length === 1,
     );
 
-    const dbToVerify = defaultDb ?? connection.databases[0];
+    const dbToVerify = defaultDb ?? databases[0];
     const isSchemalessDb =
       dbToVerify?.schemas.some((schema) => isSchemaless(schema.name)) ?? false;
 
     // For schemaless databases, treat databases as schemas
     if (isSchemalessDb) {
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      const dbToTablesMap: Record<string, any> = {};
-
-      for (const db of connection.databases) {
+      for (const db of databases) {
         const isDefaultDb = db.name === defaultDb?.name;
-
-        for (const schema of db.schemas) {
-          for (const table of schema.tables) {
-            const columns = table.columns.map((col) => col.name);
-
-            if (isDefaultDb) {
-              // For default database, add tables directly to top level
-              dbToTablesMap[table.name] = columns;
-            } else {
-              // Otherwise nest under database name
-              dbToTablesMap[db.name] = dbToTablesMap[db.name] || {};
-              dbToTablesMap[db.name][table.name] = columns;
-            }
+        const tables = db.schemas.flatMap((schema) => schema.tables);
+        builder.addDatabase([db.name], db);
+
+        for (const table of tables) {
+          if (isDefaultDb) {
+            // For default database, add tables directly to top level
+            builder.addTable([], table);
+          } else {
+            // Otherwise nest under database name
+            builder.addTable([db.name], table);
           }
         }
       }
 
       return {
         shouldAddLocalTables: false,
-        schema: dbToTablesMap,
+        schema: builder.build(),
         defaultSchema: defaultDb?.name,
       };
     }
 
-    // For default db, we can use the schema name directly
-    for (const schema of defaultDb?.schemas ?? []) {
-      schemaMap[schema.name] = {};
-      for (const table of schema.tables) {
-        const columns = table.columns.map((col) => col.name);
-        schemaMap[schema.name][table.name] = columns;
+    // For default db, we can use the schema name directly so add them to the top level
+    if (defaultDb) {
+      for (const schema of defaultDb.schemas) {
+        builder.addSchema([schema.name], schema);
+
+        for (const table of schema.tables) {
+          builder.addTable([schema.name], table);
+        }
       }
     }
 
     // Otherwise, we need to use the fully qualified name
-    for (const database of connection.databases) {
-      if (database.name === defaultDb?.name) {
```

Comment:
this is the only functional change. we now include the default db which can help with autocomplete or if users want fully qualified names